README.md

```diff
@@ -227,23 +227,56 @@ You will need to adapt the `slurm/train.slurm` script to match your cluster.
 
 #### 👨‍💻 Training with a code interpreter
 
-We provide a `code` reward function for executing code generated by the policy during training. Currently, this reward function targets code contests like [Codeforces](https://codeforces.com), where solutions are executed against a set of test cases and the overall success rate is returned as the final reward. To ensure safe execution, we use [E2B](https://e2b.dev) sandboxes, which are fast and cheap to run. To use this reward function, first install the necessary dependencies:
+We provide a `code` reward function for executing code generated by the policy during training. Currently, this reward function targets code contests like [Codeforces](https://codeforces.com), where solutions are executed against a set of test cases and the overall success rate is returned as the final reward. To ensure safe execution, we support multiple sandbox providers:
+
+1. [E2B](https://e2b.dev) - Fast, cloud-based sandboxes with focus on Python execution
+2. [Morph](https://cloud.morph.so/web/) - Cloud-based sandboxes with broader language support - Python/JS/C++/Rust
+
+To use the code reward function, first install the necessary dependencies:
 
 ```shell
 uv pip install -e '.[code]'
 ```
 
-Then create a `.env` file and place an API token from E2B within it:
+##### E2B Provider
+
+To use E2B sandboxes, create a `.env` file and add your E2B API token:
 
 ```
 E2B_API_KEY="e2b_xxx"
 ```
 
-Then make sure your dataset contains a `verification_info` column with the following schema (adopted from PrimeIntellect's excellent [datasets](https://huggingface.co/collections/PrimeIntellect/synthetic-1-67a2c399cfdd6c9f7fae0c37) of verifiable problems):
+##### Morph Provider
+
+To use Morph, first install the morphcloud package:
+
+```shell
+pip install morphcloud
```

Comment:
WDYT about adding this to the `code` extra in `setup.py` so users can just run `uv pip install -e '.[code]'`?